src/commands/exportService.ts

```diff
@@ -0,0 +1,152 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as yaml from 'js-yaml';
+import * as path from 'path';
+import { ProgressLocation, Uri, window } from "vscode";
+import { IActionContext } from 'vscode-azureextensionui';
+import * as Constants from '../constants';
+import { ApiTreeItem } from '../explorer/ApiTreeItem';
+import { ServiceTreeItem } from '../explorer/ServiceTreeItem';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { azUtils } from '../utils/azUtils';
+import { cpUtils } from '../utils/cpUtils';
+import { FileDownloader } from '../utils/FileDownloader';
+import { askFolder } from '../utils/vscodeUtils';
+
+export async function exportService(context: IActionContext, node?: ServiceTreeItem): Promise<void> {
+    if (!node) {
+        node = <ServiceTreeItem>await ext.tree.showTreeItemPicker(ServiceTreeItem.contextValue, context);
+    }
+
+    await extract(node);
+}
+
+export async function exportAPI(context: IActionContext, node?: ApiTreeItem): Promise<void> {
+    if (!node) {
+        node = <ApiTreeItem>await ext.tree.showTreeItemPicker(ApiTreeItem.contextValue, context);
+    }
+
+    const apiName = node.apiContract.name === undefined ? "" : node.apiContract.name;
+
+    await extract(node, apiName);
+}
+
+async function extract(node: ApiTreeItem | ServiceTreeItem, apiName?: string): Promise<void> {
+
+    const uris = await askFolder("Extract");
+    const sourceApimName = node.root.serviceName;
+    const resourceGroup = node.root.resourceGroupName;
+    const subscriptionId = node.root.subscriptionId;
+    const templatesFolder = await createTemplatesFolder(uris, sourceApimName);
+    const noticeContent = await generateExtractConfig(templatesFolder, sourceApimName, apiName);
+
+    window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("Extracting", noticeContent),
+            cancellable: false
+        },
+        async () => {
+            await azUtils.checkAzInstalled();
+            await runExtractor(templatesFolder, sourceApimName, resourceGroup, subscriptionId);
+        }
+    ).then(
+        () => {
+            window.showInformationMessage(localize("Extracted", `Extraction completed!`));
+        });
+}
+
+async function createTemplatesFolder(uris: Uri[], sourceApimName: string): Promise<string> {
```

Comment:
looks like you need just a single Uri here, not an array

Reply:
Yes - I have reworked this to only return a single Uri